### PR TITLE
CAMS-193 Suggested refactoring for backend

### PR DESCRIPTION
**Pull request by magile32**
# Purpose

Refactoring backend for clarity

# Major Changes

- Removed unused node package `pine`
- Logic able to handle request to `api/cases` to fetch court cases
- Rename some references to Chapter15 to reflect module and function responsibility
- Removed unused Chapter 15 logic

# Testing/Validation

- Ensure all automated tests are passing and code coverage stays above 90%
- Verified that request to `api/cases` return both Chapter 15 and Chapter 12 
- Manually verified in branch deploymnt

# Notes

### CODE REVIEW

File: backend/functions/cases/cases.function.ts [hunk 1]
```diff
@@ -21,26 +22,33 @@ const httpTrigger: AzureFunction = async function (
   functionContext: Context,
   casesRequest: HttpRequest,
 ): Promise<void> {
+  // Setup dependencies
   const applicationContext = await applicationContextCreator(functionContext);
   const casesController = new CasesController(applicationContext);
-  let caseChapter = '';
+
+  // Process request message
+  const caseId = casesRequest?.params?.caseId;
```
Comment (fmaddenflx) [inline backend/functions/cases/cases.function.ts:30]: Is caseId now required?  By default, if none is provided, then caseId will be undefined.  Is that what we want?

Comment (fmaddenflx) [reply backend/functions/cases/cases.function.ts:30]: Looks like that would be fine.

File: backend/functions/lib/adapters/gateways/cases.dxtr.gateway.test.ts [hunk 2]
```diff
@@ -70,6 +70,56 @@ describe('Test DXTR Gateway', () => {
...
+      value: dateFiledFrom,
+    };
+    expect(querySpy).toHaveBeenCalledWith(
+      expect.anything(),
+      appContext.config.dxtrDbConfig,
+      expect.anything(),
+      expect.arrayContaining([expect.objectContaining(expectedDateInput)]),
+    );
+    expect(actualResult).not.toEqual(cases);
+    console.log(actualResult);
```
Comment (fmaddenflx) [inline backend/functions/lib/adapters/gateways/cases.dxtr.gateway.test.ts:121]: probably remove this console log

File: backend/functions/lib/adapters/controllers/cases.controller.ts [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
     this.context = context;
+    this.courtCaseManagement = new CourtCaseManagement();
   }
 
   public async getCaseList(requestQueryFilters: { caseChapter: string }) {
     log.info(this.context, MODULE_NAME, 'Getting case list.');
     if (requestQueryFilters.caseChapter == '15') {
-      const chapter15CaseList = new Chapter15CaseList();
-      return await chapter15CaseList.getChapter15CaseList(this.context);
+      return await this.courtCaseManagement.getChapter15CaseList(this.context);
```
Comment (fmaddenflx) [inline backend/functions/lib/adapters/controllers/cases.controller.ts:20]: Should we also change this method name getChapter15CaseList ?

File: backend/functions/lib/adapters/gateways/cases.dxtr.gateway.ts [hunk 4]
```diff
@@ -37,9 +37,46 @@ export default class CasesDxtrGateway implements CasesInterface {
   async getChapter15Cases(
     context: ApplicationContext,
     options: { startingMonth?: number },
-  ): Promise<Chapter15CaseInterface[]> {
-    const doChapter12Enable = context.featureFlags['chapter-twelve-enabled'];
-    const rowsToReturn = doChapter12Enable ? '10' : '20';
```
Comment (fmaddenflx) [inline backend/functions/lib/adapters/gateways/cases.dxtr.gateway.ts:42]: Are we no longer using the chapter-twelve-enabled flag?  And what about the number of rows to return?

File: backend/functions/cases/cases.function.ts (hunk 1 above)
Comment (magile32) [reply backend/functions/cases/cases.function.ts:30]: CaseId is not required. Logic will try to grab that value but if the caseId is not set it doesn't return the case details.

File: backend/functions/lib/adapters/controllers/cases.controller.ts (hunk 3 above)
Comment (magile32) [reply backend/functions/lib/adapters/controllers/cases.controller.ts:20]: As of now, the logic only fetches chapter 15 course cases, so I let be for now. We could discuss this further if we want to change or remove this.

File: backend/functions/lib/adapters/gateways/cases.dxtr.gateway.ts (hunk 4 above)
Comment (magile32) [reply backend/functions/lib/adapters/gateways/cases.dxtr.gateway.ts:42]: I noticed this too and this should not be. I must have inadvertently checked in changes I didn't mean to or something went wrong with a rebase. This should be fixed now.

File: backend/functions/lib/adapters/controllers/cases.controller.ts (hunk 3 above)
Comment (btposey) [reply backend/functions/lib/adapters/controllers/cases.controller.ts:20]: As stated in stand-up, I'm sympathetic to generic naming, but Mike is right in that this is highly chapter 15 specific right now. Call YAGNI for now until future refactors are more chapters are supported.

**Review by btposey — APPROVED** [review]

**Review by jamesobrooks — APPROVED** [review]

**Review by fmaddenflx — APPROVED** [review]

**Pull request merged**